app/services/catalog/submit_order.rb

```diff
@@ -22,9 +22,11 @@ def process
     private
 
     def submit_order_item(item)
-      TopologicalInventory.call do |api_instance|
-        result = api_instance.order_service_plan(item.service_plan_ref, parameters(item))
-        update_item(item, result)
+      ManageIQ::API::Common::Request.with_request(item.context.transform_keys(&:to_sym)) do
```

Comment:
@lindgrenj6 I think the with_request should be in the transition class so we leave this one alone so it can be called from other places otherwise we could be setting the headers twice.